### ISSUE TITLE
SC:BW crashes when start a game with BWEnv.dll

### ISSUE THREAD

**Issue by bluecontra** [TorchCraft/TorchCraft#235]
Environment:
Server: Windows7, StarCraft 1.16.1, BWAPI 4.2.0 (4.1.2 also tried)

Client: Ubuntu 16.04 LTS, Torch7, TorchCraft 1.3.0 (other versions also tried)

Issues:
I follow the instructions and my installation of client on Linux and server on Windows goes smoothly.
But everytime I try to start a game through Chaoslauncher with the BWEnv.dll copied from TorchCraft release (several versions are tried, same issue), the game crashes at very beginning (maybe the loading process) as the screenshot shown below:
![default](https://user-images.githubusercontent.com/17591106/33753241-495ef134-dc21-11e7-9f3c-7235a32eac04.png)

Other things:
1. I checked the previous issues raised and found that #120 #201 are similar ones.
I checked my error log and it is almost the same as it in #120 , but no further and clear information about how to fix this.
In #201 , the issue seems to be a little bit different,  because my game crashes at earlier phase, maybe loading phase while crash occurs at the inital frame in 201's case.  And I tried the 4.1.2 BWAPI and it didn't work.
2. I also think it may be something wrong with BWEnv.dll, becasue similarly the game can run smoothly when I don't point the _ai = BWEnv.dll_.
3. If I leave the server game in the crashed state, and then run the client on Linux through _python py/example.py -t $ip$_, on the server (Windows) side, SC:BW is aborted due to some fatal error and on client (Linux) side,  'Nonetype' error occurs in _example.py_ near:
`    for pid, player in state.player_info.items():
        print("player {} named {} is {}".format(player.id, player.name,
tc.Constants.races._dict[player.race]))` 
Is this means the server returns NULL state information and client cannot get unit information?

Anybody can help me with this issue? 
Really want to know how to make it work. Thanks in advance.

**Comment by ebetica** [comment]
If you hit the stuck, empty window, then your server is most likely waiting for a client to connect. You can verify by checking if anything is listening to port 11111 on Windows be in the "crashed" state. 

Most likely, the server is working as intended and there's a bug either in the communication or the client. I encourage you to use a debugger in step 3 and see what values are being sent.

**Comment by bluecontra** [comment]
@ebetica Thanks for your help.
I think I eventually found where the issue happens. It is the problem in client instead of the server. I tried the alibaba/torchcraft-py as a python client and it works fine. 
I think maybe something went wrong with the original python client script `py/example.py` (but I haven't take a check in it yet).

Anyway, thanks a lot. Any further feedback will be attached here. : )
